src/replica/replica_stub.cpp

```diff
@@ -1986,11 +1971,14 @@ void replica_stub::open_replica(const app_info &app,
 
 ::dsn::task_ptr replica_stub::begin_close_replica(replica_ptr r)
 {
-    dassert(r->status() == partition_status::PS_ERROR ||
-                r->status() == partition_status::PS_INACTIVE,
-            "%s: invalid state %s when calling begin_close_replica",
-            r->name(),
-            enum_to_string(r->status()));
+    dassert_f(r->status() == partition_status::PS_ERROR ||
+                  r->status() == partition_status::PS_INACTIVE ||
+                  r->disk_migrator()->status() >= disk_migration_status::MOVED,
+              "invalid state(partition_status={}, migration_status={}) when calling "
+              "replica({}) close",
+              enum_to_string(r->status()),
+              enum_to_string(r->disk_migrator()->status()),
+              r->name());
```

Comment:
I think this check is redundant ? You already add this check at replica::close.

Reply:
`replica->close` is called by `replica_stub->close`